lib/HTTP/EpaperWebServer.cpp

```diff
@@ -242,6 +243,9 @@ void EpaperWebServer::handleFirmwareUpdateUpload(RequestContext& request) {
       } else {
         this->updateSuccessful = true;
       }
+    } else {
+      Serial.print(request.upload.index / (Update.size() / 100) + 1);
+      Serial.println(F("%"));
```

Comment:
This is a great idea, however it's spammy enough that it slows down the update a bit if the serial console is open.

Static function variable to keep track of the progress from the last instantiation works nicely:

```diff
 void EpaperWebServer::handleFirmwareUpdateUpload(RequestContext& request) {
+  // Keep track of upload progress so we can print throttled updates to Serial
+  static uint8_t uploadProgress;
+
   if (request.upload.index == 0) {
+    uploadProgress = 0;
+
     // Give up if the filename starts with "INITIALIZER_".  These binary images
     // are built by custom platformio tooling that includes all parts of flash
     // necessary get started (including bootloader and partition table).
@@ -244,8 +249,12 @@ void EpaperWebServer::handleFirmwareUpdateUpload(RequestContext& request) {
         this->updateSuccessful = true;
       }
     } else {
-      Serial.print(request.upload.index / (Update.size() / 100) + 1);
-      Serial.println(F("%"));
+      const uint8_t newProgress = (request.upload.index / (Update.size() / 100) + 1);
+
+      if (newProgress != uploadProgress) {
+        Serial.printf_P(PSTR("%d%%\n"), newProgress);
+        uploadProgress = newProgress;
+      }
     }
```

Reply:
:+1: